Sources/SwiftDocC/Model/ParametersAndReturnValidator.swift

```diff
@@ -0,0 +1,473 @@
+            guard hasDocumentedReturnValues, Self.shouldAddObjectiveCErrorParameter(signatures, parameters) else {
+                // The symbol has no parameter documentation or return value documentation and none should be synthesized.
+                return DocumentationDataVariants(defaultVariantValue: nil)
+            }
+            // Since the symbol documented its return value synthesize an "error" parameter for it.
+            var variants = DocumentationDataVariants<ParametersSection>()
+            for trait in signatures.keys {
+                variants[trait] = ParametersSection(parameters: []) // Add an empty section so that this language doesn't fallback to the Objective-C content.
+            }
+            variants[.objectiveC] = ParametersSection(parameters: [Parameter(name: "error", contents: Self.objcErrorDescription)]) // This parameter is synthesized and doesn't have a source range.
+            return variants
+        }
+        
+        var variants = DocumentationDataVariants<ParametersSection>()
+        var allKnownParameterNames: Set<String> = []
+        var argumentLabelsAndPossibleParameterNames: [String: Set<String>] = [:]
+        // Group the parameters by name to be able to diagnose parameters documented more than once.
+        let parametersByName = [String: [Parameter]](grouping: parameters, by: \.name)
+        
+        for (trait, signature) in signatures {
+            // Collect all language representations's parameter information from the function signatures.
+            let knownParameterNames = Set(signature.parameters.map(\.name))
+            allKnownParameterNames.formUnion(knownParameterNames)
+            for parameter in signature.parameters {
+                if let externalName = parameter.externalName {
+                    argumentLabelsAndPossibleParameterNames[externalName, default: []].insert(parameter.name)
+                }
+            }
+            
+            // Remove documented parameters that don't apply to this language representation's function signature.
+            var languageApplicableParameters = parametersByName.filter { knownParameterNames.contains($0.key) }
+            guard !languageApplicableParameters.isEmpty else { 
+                // Add an empty section so that this language doesn't fallback to another language's content.
+                variants[trait] = ParametersSection(parameters: [])
+                continue
+            }
+            
+            // Add a missing error parameter documentation if needed.
+            if trait == .objectiveC, Self.shouldAddObjectiveCErrorParameter(signatures, parameters) {
+                languageApplicableParameters["error"] = [Parameter(name: "error", contents: Self.objcErrorDescription)] // This parameter is synthesized and doesn't have a source range.
+            }
+            
+            // Ensure that the parameters are displayed in the order that they're declared.
```

Comment:
Why? Shouldn't authors have the flexibility to display parameters in any order?

Reply:
My reasoning was that the parameters need to be passed in a specific order so I thought that it was best to ensure that they're also listed in that order.

Reply:
I updated this documentation in https://github.com/apple/swift-docc/pull/776/commits/ef264671245be85daf65bb1ab7a3eb96a6d8a95b to clarify that it's the order that the developer needs to pass the parameters